tests/backends/launcher.py

```diff
@@ -0,0 +1,218 @@
+        err = str(err.decode("utf-8"))
+        if 'Exception' in err or 'Error' in err:
+            raise Exception(err)
+    except TimeoutExpired:
+        p.kill()
+        std, err = p.communicate()
+    return std, err
+
+
+def create_cmd_lines(cmd_line, **kwargs):
+    keys = sorted(kwargs.keys())
+    values_comb = itertools.product(*(kwargs[k] for k in keys))
+    cmd_lines = []
+    for combi in values_comb:
+        temp_cmd_line = cmd_line[::]
+        for key_idx, v in enumerate(combi):
+            k = keys[key_idx]
+            temp_cmd_line = temp_cmd_line.replace(f"[{k}]", str(v))
+        cmd_lines.append(temp_cmd_line)
+    cmd_lines = list(set(cmd_lines))
+    return cmd_lines
+
+
+def undecorated(o):
+    """Remove all decorators from a function, method or class"""
+    # class decorator
+    if type(o) is type:
+        return o
+
+    try:
+        closure = o.__closure__
+    except AttributeError:
+        return
+
+    if closure:
+        for cell in closure:
+            # avoid infinite recursion
+            if cell.cell_contents is o:
+                continue
+
+            # check if the contents looks like a decorator; in that case
+            # we need to go one level down into the dream, otherwise it
+            # might just be a different closed-over variable, which we
+            # can ignore.
+
+            # Note: this favors supporting decorators defined without
+            # @wraps to the detriment of function/method/class closures
+            if looks_like_a_decorator(cell.cell_contents):
+                undecd = undecorated(cell.cell_contents)
+                if undecd:
+                    return undecd
+        else:
+            return o
+    else:
+        return o
+
+
+def looks_like_a_decorator(a):
+    return (
+        isfunction(a) or ismethod(a) or isclass(a)
+    )
+
+
+class DDPLauncher:
+    """
+    This script is used to launch DDP related tests.
+    It provides a simple decorator to run your test. See below for explication and example:
+
+    Example:
+
+        # The decorator will read cmd_line + arguments provided as kwargs.
+
+        @DDPLauncher.run("--max_epochs [max_epochs] --gpus 2 --accelerator [accelerator]",
+                         max_epochs=["1"],
+                         accelerator=["ddp", "ddp_spawn"])
+        def test_cli_to_pass(tmpdir, args=None):
+
+            ... do something with args + BoringModel
+
+            return '1'
+
+
+    Explication:
+
+        1 - DDPLauncher.run will recieve a command line to run where tokens are recognized by [].
+            DDPLauncher.run will emulate pytest `parametrize` function and generate as many
+            resolved cmd_lines from your provided cmd_line + product of your kwargs arguments.
+            For the previous example, it will generate 2 cmd_lines and run them:
+                1: --max_epochs 1 --gpus 2 --accelerator ddp
+                2: --max_epochs 1 --gpus 2 --accelerator ddp_spawn
```

Comment:
well as a user writing test I do not care about CLI arguments, I just want to define what arguments/values shall be tested as it is now that all combination is created, there is an extra resin for define string CLI, right?